posthog/temporal/data_imports/pipelines/google_ads/source.py

```diff
@@ -322,6 +322,11 @@ def get_rows() -> collections.abc.Iterator[pa.Table]:
         name=name,
         items=get_rows(),
         primary_keys=table.primary_key,
+        partition_count=1 if table.requires_filter else None,  # this enables partitioning
+        partition_size=1 if table.requires_filter else None,  # this enables partitioning
+        partition_mode="datetime" if table.requires_filter else None,
+        partition_format="day" if table.requires_filter else None,
+        partition_keys=["segments_date"] if table.requires_filter else None,
```

Comment:
eventually, we could have this be configured by the segment the user chooses, but at the time we don't have support in the UI to customize schemas, so this works :+1: